drivers/video/isx019.c

```diff
@@ -0,0 +1,3247 @@
+    }
+
+  i2c_unlock();
+  return ret;
+}
+
+static int send_read_cmd(struct i2c_config_s *config,
+                          uint8_t cat,
+                          uint16_t addr,
+                          uint8_t size)
+{
+  int ret;
+  int len;
+  uint8_t buf[ISX019_I2C_RDREQ_TOTALLEN] =
+    {
+      0
+    };
+
+  buf[ISX019_I2C_OFFSET_TOTALLEN]  = ISX019_I2C_RDREQ_TOTALLEN;
+  buf[ISX019_I2C_OFFSET_CMDNUM]    = 1;
+  buf[ISX019_I2C_OFFSET_CMDLEN]    = ISX019_I2C_RDREQ_LEN;
+  buf[ISX019_I2C_OFFSET_CMD]       = ISX019_I2C_CMD_READ;
+  buf[ISX019_I2C_OFFSET_CATEGORY]  = cat;
+  buf[ISX019_I2C_OFFSET_ADDRESS_H] = addr >> 8;
+  buf[ISX019_I2C_OFFSET_ADDRESS_L] = addr & 0xff;
+  buf[ISX019_I2C_OFFSET_READSIZE]  = size;
+
+  len = ISX019_I2C_OFFSET_READSIZE + 1;
+  buf[len] = calc_isx019_chksum(buf, len);
+  len++;
+
+  ret = i2c_write(g_isx019_private.i2c, config, buf, len);
+  return ret;
+}
+
+static int isx019_i2c_read(uint8_t  cat,
+                           uint16_t addr,
+                           uint8_t  *data,
+                           uint8_t  size)
+{
+  int ret;
+  struct i2c_config_s config;
+
+  DEBUGASSERT(size <= ISX019_I2C_REGSIZE_MAX);
+
+  config.frequency = ISX019_I2C_FREQUENCY;
+  config.address   = ISX019_I2C_SLVADDR;
+  config.addrlen   = ISX019_I2C_SLVADDR_LEN;
+
+  i2c_lock();
+
+  ret = send_read_cmd(&config, cat, addr, size);
+  if (ret == OK)
+    {
+      ret = recv_read_response(&config, data, size);
+    }
+
+  i2c_unlock();
+
+  return ret;
+}
+
+static void fpga_init(void)
+{
+  uint8_t regval;
+
+  regval = FPGA_RESET_ENABLE;
+  fpga_i2c_write(FPGA_RESET, &regval, 1);
+  regval = FPGA_DATA_OUTPUT_STOP;
+  fpga_i2c_write(FPGA_DATA_OUTPUT, &regval, 1);
+  fpga_activate_setting();
+  regval = FPGA_RESET_RELEASE;
+  fpga_i2c_write(FPGA_RESET, &regval, 1);
+  fpga_activate_setting();
+}
+
+static int set_drive_mode(void)
+{
+  uint8_t drv[] =
+    {
+#ifdef CONFIG_VIDEO_ISX019_DOL2
+      DOL2_30FPS_SENS, DOL2_30FPS_POST, DOL2_30FPS_SENSPOST, DOL2_30FPS_IO
+#else
+      DOL3_30FPS_SENS, DOL3_30FPS_POST, DOL3_30FPS_SENSPOST, DOL3_30FPS_IO
+#endif
+    };
+
+  nxsig_usleep(TRANSITION_TIME_TO_STARTUP);
+
+  isx019_i2c_write(CAT_CONFIG, MODE_SENSSEL,      &drv[INDEX_SENS], 1);
+  isx019_i2c_write(CAT_CONFIG, MODE_POSTSEL,      &drv[INDEX_POST], 1);
+  isx019_i2c_write(CAT_CONFIG, MODE_SENSPOST_SEL, &drv[INDEX_SENSPOST], 1);
+
+  nxsig_usleep(TRANSITION_TIME_TO_STREAMING);
+
+  return OK;
+}
+
+static bool try_repeat(int sec, int usec, int (* trial_func)(void))
+{
+  int ret;
+  struct timeval start;
+  struct timeval now;
+  struct timeval delta;
+  struct timeval wait;
+
+  wait.tv_sec = sec;
+  wait.tv_usec = usec;
+
+  gettimeofday(&start, NULL);
+  while (1)
+    {
+      ret = trial_func();
+      if (ret != -ENODEV)
+        {
+          break;
+        }
+      else
+        {
+          gettimeofday(&now, NULL);
+          timersub(&now, &start, &delta);
+          if (timercmp(&delta, &wait, >))
```

Comment:
I think most probably `if (timercmp(&delta, &wait, > ))` should work. But I see that we have similar case in `apps/examples/camera/camera_main.c`

Reply:
@pkarashchenko 
Thanks for the quick reply 
Sadly the suggested fix will not work because we have space before ) and nxstyle will throw an error.

My question was actually about the failed macOS build

Reply:
:) Sorry, I misunderstood your question

Reply:
I think that ESP32 build on MAC issue was resolved yesterday. Please rebase and check.